integration/tests/access/cohort3/execution_data_pruning_test.go

```diff
@@ -0,0 +1,270 @@
+	accessNode := s.net.ContainerByName(s.accessNodeName)
+	observerNode := s.net.ContainerByName(s.observerNodeName)
+
+	waitingBlockHeight := uint64(200)
+	s.waitUntilExecutionDataForBlockIndexed(waitingBlockHeight)
+	s.net.StopContainers()
+
+	metrics := metrics.NewNoopCollector()
+
+	// start an execution data service using the Access Node's execution data db
+	anEds := s.nodeExecutionDataStore(accessNode)
+
+	// setup storage objects needed to get the execution data id
+	anDB, err := accessNode.DB()
+	require.NoError(s.T(), err, "could not open db")
+
+	anHeaders := storage.NewHeaders(metrics, anDB)
+	anResults := storage.NewExecutionResults(metrics, anDB)
+
+	// start an execution data service using the Observer Node's execution data db
+
+	onEds := s.nodeExecutionDataStore(observerNode)
+	// setup storage objects needed to get the execution data id
+	onDB, err := observerNode.DB()
+	require.NoError(s.T(), err, "could not open db")
+
+	onResults := storage.NewExecutionResults(metrics, onDB)
+
+	s.checkResults(anHeaders, anResults, onResults, anEds, onEds)
+}
+
+// waitUntilExecutionDataForBlockIndexed waits until the execution data for the specified block height is indexed.
+func (s *ExecutionDataPruningSuite) waitUntilExecutionDataForBlockIndexed(waitingBlockHeight uint64) {
+	observerNode := s.net.ContainerByName(s.observerNodeName)
+
+	grpcClient, err := s.getGRPCClient(observerNode.Addr(testnet.GRPCPort))
+	s.Require().NoError(err)
+
+	// creating execution data api client
+	client, err := getClient(fmt.Sprintf("localhost:%s", observerNode.Port(testnet.ExecutionStatePort)))
+	s.Require().NoError(err)
+
+	// pause until the observer node start indexing blocks
+	s.Require().Eventually(func() bool {
+		_, err := grpcClient.GetEventsForHeightRange(s.ctx, &accessproto.GetEventsForHeightRangeRequest{
+			Type:                 sdk.EventAccountCreated,
+			StartHeight:          1,
+			EndHeight:            1,
+			EventEncodingVersion: entities.EventEncodingVersion_CCF_V0,
+		})
+
+		return err == nil
+	}, 2*time.Minute, 10*time.Second)
+
+	// subscribe on events till waitingBlockHeight to make sure that execution data for block indexed till waitingBlockHeight and pruner
+	// pruned execution data at least once
+	// SubscribeEventsFromStartHeight used as subscription here because we need to make sure that execution data are already indexed
+	stream, err := client.SubscribeEventsFromStartHeight(s.ctx, &executiondata.SubscribeEventsFromStartHeightRequest{
+		EventEncodingVersion: entities.EventEncodingVersion_CCF_V0,
+		Filter:               &executiondata.EventFilter{},
+		HeartbeatInterval:    1,
+		StartBlockHeight:     0,
+	})
+	s.Require().NoError(err)
+	eventsChan, errChan, err := SubscribeHandler(s.ctx, stream.Recv, eventsResponseHandler)
+	s.Require().NoError(err)
+
+	for {
+		select {
+		case err := <-errChan:
+			s.Require().NoErrorf(err, "unexpected %s error", s.observerNodeName)
+		case event := <-eventsChan:
```

Comment:
can you add a timeout as well.

Reply:
[0cba0a9](https://github.com/onflow/flow-go/pull/6217/commits/0cba0a9ff7fa03f438e75efc5b93223ad8fc3089)